static/js/src/advantage/react/utils/copyToClipboardAsync.ts

```diff
@@ -0,0 +1,37 @@
+function fallbackCopyToClipboard(text: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const textArea = document.createElement("textarea");
+    textArea.value = text;
+
+    // Avoid scrolling to the bottom
+    textArea.style.top = "0";
+    textArea.style.left = "0";
+    textArea.style.position = "fixed";
+
+    document.body.appendChild(textArea);
+    textArea.focus();
+    textArea.select();
+
+    try {
+      const successful = document.execCommand("copy");
+
+      if (successful) {
+        resolve();
+      } else {
+        reject(new Error("Failed to copy text"));
+      }
+    } catch {
+      reject(new Error("Exception during copying text"));
+    } finally {
+      document.body.removeChild(textArea);
+    }
+  });
+}
+
+export function copyToClipboardAsync(text: string) {
+  if (!navigator.clipboard) {
+    return fallbackCopyToClipboard(text);
+  }
```

Comment:
What are the cases navigator.clipboard don't exist? just wondering

Reply:
@minkyngkm `navigator.clipboard` is only available in secure contexts (i.e. HTTPS). It doesn't work on localhost unless you manually add https in the URL. I added a fallback so it works on http as well.

I guess this can be removed since u.com will never be on http, what do you think?

Reply:
@fasih-mehmood yep. it seems u.com always redirects to https.

Reply:
Alright, I'll remove the fallback then.